### PR TITLE
Add audit log to sidecar injector

### PR DESCRIPTION
**Pull request by wcs1only**
# Description

Emit a log line on sidecar injection success or failure. 
## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #3009 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/injector/injector.go [hunk 1]
```diff
@@ -212,6 +212,7 @@ func (i *injector) handleRequest(w http.ResponseWriter, r *http.Request) {
 
 	if err != nil {
 		admissionResponse = toAdmissionResponse(err)
+		log.Errorf("Sidecar injector failed to inject for app '%s'", diagAppID)
```
Comment (artursouza) [inline pkg/injector/injector.go:215]: Please, add error into message.

File: pkg/injector/injector.go [hunk 2]
```diff
@@ -254,12 +255,14 @@ func (i *injector) handleRequest(w http.ResponseWriter, r *http.Request) {
 			http.StatusInternalServerError,
 		)
 
+		log.Errorf("Sidecar injector failed to inject for app '%s'", diagAppID)
```
Comment (artursouza) [inline pkg/injector/injector.go:258]: same.

File: pkg/injector/injector.go [hunk 3]
```diff
@@ -254,12 +255,14 @@ func (i *injector) handleRequest(w http.ResponseWriter, r *http.Request) {
...
 		monitoring.RecordFailedSidecarInjectionCount(diagAppID, "response")
 	}
 	w.Header().Set("Content-Type", "application/json")
 	if _, err := w.Write(respBytes); err != nil {
 		log.Error(err)
 	} else {
+		log.Infof("Sidecar injector succeeded injection for app '%s'", diagAppID)
```
Comment (artursouza) [inline pkg/injector/injector.go:265]: same.

**Review by artursouza — CHANGES_REQUESTED** [review]

File: pkg/injector/injector.go (hunk 2 above)
Comment (wcs1only) [reply pkg/injector/injector.go:258]: We do already at 250, but I guess it should be one log line. Will fix.

**Review by artursouza — APPROVED** [review]

**Pull request merged**